### DISCUSSION TITLE
Hardware acceleration of PDF.js canvas

### DISCUSSION THREAD

**Discussion by shivaprsd** [mozilla/pdf.js#18199] [Q&A] [answered]
Hi devs, I recently noticed that newer versions of Firefox (macOS) enable hardware acceleration of HTML Canvas by default. Consequently, PDF.js rendered canvases are GPU accelerated.

Is there any performance benefit to this in the rendering pipeline of a typical PDF? More importantly, is there any significant drawbacks to disabling the canvas HW acceleration?

As I understand it, HW acceleration is indispensable only to resource-intensive graphics applications.

**Comment by marco-c** [comment, marked as answer]
There are pros and cons, we are evaluating if it makes sense to keep HW acceleration enabled or disable it for PDF.js.

**Comment by marco-c** [comment]
Example case where it is bad: https://bugzilla.mozilla.org/show_bug.cgi?id=1895499.

**Comment by shivaprsd** [comment]
@marco-c I don't know if it's the same as the example case, but even for the demo PDF (`compressed.tracemonkey-pldi-09.pdf`) it is apparently bad on p. 11:

With HWA | Without HWA
--- | ---
![with-HWA](https://github.com/mozilla/pdf.js/assets/18247990/6ab480bd-3ecb-4c57-b318-379967b6198a) | ![without-HWA](https://github.com/mozilla/pdf.js/assets/18247990/85c20569-902e-4b16-a357-37a2f0e29614)

**Reply by marco-c** [reply]
Could you share details about your machine? Which OS, gfx card, CPU, etc.

**Reply by shivaprsd** [reply]
Sure:

- machine: MacBook Air M2, 8GB, 8-core GPU
- OS: macOS Ventura 13.5 (22G74)
- Firefox: Developer Edition 125.0b9

Do you think I should open an issue?

**Reply by marco-c** [reply]
@shivaprsd I've opened a bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1901708. Thanks!
